### ISSUE TITLE
Aggressive caching; far too difficult to refresh updated assets

### ISSUE THREAD

**Issue by MarcusLlewellyn** [highfidelity/hifi#12598]
As a creator, it is very common to upload, make changes, and re-upload. This iteration can happen many, many times. There needs to be some way to refresh an object in the client. I'm aware that there is a kludgy way to do this if you're serving assets via HTTP, but this really isn't acceptable.

**Comment by sethalves** [comment]
Which file-types are causing this bad experience?  Model files?  Scripts?  Shaders?  (all?)

**Comment by Richardus** [comment]
Scripts have a reload button. SO fra not seen problems with that. The problems i have are with.

-model url
-compound url

**Comment by L370** [comment]
By placing a ? at the end of the filename, this will force the cache to expire that item and reload it from the source.

**Comment by MarcusLlewellyn** [comment]
Yes, and the "cache busting" method of using a query that is otherwise ignored is a crude workaround. As a creator, I have to keep track of what values I used to bust the cache or do something like generate unique UUIDs. I really shouldn't have to do either of these things.

**Comment by Richardus** [comment]
Marcus, thanks for confirming. I where alrerady thinking. what's new with ? compared to ?v= <type something> thatwe used for long time. and it's terrible because you lose rack of numbering and triggering old cache version.

The system need to handle something basic like this.
